src/_pytest/config/__init__.py

```diff
@@ -642,7 +642,9 @@ def add_cleanup(self, func):
     def _do_configure(self):
         assert not self._configured
         self._configured = True
-        self.hook.pytest_configure.call_historic(kwargs=dict(config=self))
+        with warnings.catch_warnings():
```

Comment:
note for verification, this hook gets triggered on all following plugin registrations as well, so we might need it around plugin registration as well (but this one is a followup consideration)